tests/test_url_building.py

```diff
@@ -15,11 +15,11 @@
 URL_FOR_ARGS2 = dict(arg1=['v1', 'v2'], _anchor='anchor')
 URL_FOR_VALUE2 = '/myurl?arg1=v1&arg1=v2#anchor'
 URL_FOR_ARGS3 = dict(arg1='v1', _anchor='anchor', _scheme='http',
-                     _server='{}:{}'.format(test_host, test_port), _external=True)
-URL_FOR_VALUE3 = 'http://{}:{}/myurl?arg1=v1#anchor'.format(test_host, test_port)
+                     _server='{}:PORT_PLACEHOLDER'.format(test_host), _external=True)
```

Comment:
Why can't you just use the previous format logic? :thinking:

Reply:
To reach the concurrency goal, I created multiple server instances by each worker.
When you run it with xdist (with -n options in pytest), it will run each server instances with different port. So the port will be decided in runtime.

See tests/conftest.py how I injected the test client port. So the formatting must be delayed after running it.
Not sure but maybe it still can be solved in previous way with more knowledge about sanic & pytest. Any advice?

Reply:
I see, I don't have a better idea right now. You could do `{}:{}.format(HOST, '{}')` but I think it's too contrived.